modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -466,6 +469,12 @@ public void visit(BLangAssignment assignNode) {
             }
             ((BLangVariableReference) varRef).lhsVar = true;
             expTypes.add(typeChecker.checkExpr(varRef, env).get(0));
+            if (varRef instanceof  BLangSimpleVarRef && ((BLangVariableReference) varRef).symbol.flags == Flags.CONST
+                    && env.enclInvokable != env.enclPkg.initFunction) {
+                dlog.error(varRef.pos, DiagnosticCode.INVALID_VARIABLE_ASSIGNMENT, varRef);
```

Comment:
Let's use a meaningful error here. We've used the following error in the previous compiler.

cannot.assign.value.constant = cannot assign a value to constant ''{0}''